src/MicrosoftTeams.ts

```diff
@@ -417,25 +441,132 @@ namespace microsoftTeams
+            try
+            {
+                if (childWindow)
+                {
+                    childWindow.close();
+                }
+            }
+            finally
+            {
+                childWindow = null;
+                childOrigin = null;
+            }
+        }
+
+        function openAuthenticationWindow(url: string, width?: number, height?: number): void
+        {
+            // Close the previously opened window if we have one
+            closeAuthenticationWindow();
+
+            // Start with a sensible default size
+            width = width || 600;
+            height = height || 400;
+
+            // Ensure that the new window is always smaller than our app's window so that it never fully covers up our app
+            width = Math.min(width, (currentWindow.innerWidth - 400));
+            height = Math.min(height, (currentWindow.innerHeight - 200));
+
             // Convert any relative URLs into absolute ones before sending them over to our parent window
             let link = document.createElement("a");
-            link.href = authenticateParameters.url;
+            link.href = url;
 
-            let messageId = sendMessage(
-                "authentication.authenticate",
-                link.href,
-                authenticateParameters.width,
-                authenticateParameters.height);
-            callbacks[messageId] = (success: boolean, response: string) =>
+            if (hostClientType === hostClientTypes.desktop)
             {
-                if (success)
+                // We are running in our desktop app so give our main process a heads up on the URL we are about
+                // to call window.open with; this will allow the window.open call to pass through without getting
+                // kicked out into the browser (which is the default behavior)
+                let messageId = sendMessageRequest(parentWindow, "authentication.allowWindowOpen", link.href);
+                callbacks[messageId] = (allowed: boolean) =>
                 {
-                    authenticateParameters.successCallback(response);
+                    if (allowed)
+                    {
+                        // Open the window with a desired set of electron BrowserWindow features
+                        childWindow = currentWindow.open(link.href, "_blank", "width=" + width + ", height=" + height);
+                    }
+                    else
+                    {
+                        throw new Error("Authentication is only supported for URLs matching the pattern registered in the manifest.");
+                    }
+                };
+            }
+            else
+            {
+                // We are running in the browser so we need to center the new window ourselves
+                let left: number = (typeof currentWindow.screenLeft !== undefined) ? currentWindow.screenLeft : currentWindow.screenX;
```

Comment:
>(typeof currentWindow.screenLeft !== undefined) [](start = 35, length = 47)

the typeof operator always returns a string, so this never passes. You should use `"undefined"` (the string) instead of `undefined` (the literal).

Alternately, you can just remove the typeof operator completely and check if `currentWindow.screenLeft !== undefined`